numpy/core/src/multiarray/calculation.c

```diff
@@ -86,6 +95,27 @@ PyArray_ArgMax(PyArrayObject *op, int axis, PyArrayObject *out)
     if (ap == NULL) {
         return NULL;
     }
+
+    // Decides the shape of the output array.
+    if (!keepdims) {
+        out_ndim = PyArray_NDIM(ap) - 1;
+        out_shape = PyArray_DIMS(ap);
+    } else {
+        out_shape = _shape_buf;
+        if (axis_copy == NPY_MAXDIMS) {
+            for( int i = 0; i < out_ndim; i++ ) {
+                out_shape[i] = 1;
+            }
+        } else {
```

Comment:
NIT (also above):
```suggestion
        }
        else {
```
Is the style we use in NumPy.